### ISSUE TITLE
Only "update all" of those devices that are actually outdated 

### ISSUE THREAD

**Issue by andreas-bulling** [esphome/feature-requests#2129]
Hi,

With 30+ esphome devices in my HA installation I'd love to use the "Update all" functionality provided by the dashboard. Unfortunately, this updates really **all** devices - independent of their status. That is, even devices that already have the latest firmware installed are being updated. A smarter way of doing this, and much more useful and user-friendly, is to update only those devices for which a firmware update is available.

**Comment by lecrams** [comment]
even better is to have a pulldown menu where you can select one or more devices to be updated. 
Using the color scheme of the dashboard while displaying the devices to be selected would make it easy to identify a device

**Comment by fdcastel** [comment]
Related: #2205

**Comment by TommySharpNZ** [comment]
Yeah this seems totally random to try and update devices that are already on the latest version....

**Comment by ph0tic** [comment]
How is this not a thing still?
